include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -30,17 +39,30 @@ namespace eigensolver {
 namespace internal {
 
 template <class T>
-void copySingleTile(hpx::shared_future<matrix::Tile<const T, Device::CPU>> in,
-                    hpx::future<matrix::Tile<T, Device::CPU>> out) {
+void setZero(Matrix<T, Device::CPU>& mat) {
+  dlaf::matrix::util::set(mat, [](auto&&) { return static_cast<T>(0.f, 0.f); });
+}
+
+template <Device device, class T>
+void copySingleTile(hpx::shared_future<matrix::Tile<const T, device>> in,
+                    hpx::future<matrix::Tile<T, device>> out) {
   hpx::dataflow(dlaf::getCopyExecutor<Device::CPU, Device::CPU>(),
-                matrix::unwrapExtendTiles(matrix::copy_o), in, std::move(out));
+                matrix::unwrapExtendTiles(matrix::copy_o), in, out);
 }
 
 template <class Executor, Device device, class T>
 void trmmPanel(Executor&& ex, hpx::shared_future<matrix::Tile<const T, device>> t,
                hpx::future<matrix::Tile<T, device>> w) {
   hpx::dataflow(ex, matrix::unwrapExtendTiles(tile::trmm_o), blas::Side::Right, blas::Uplo::Upper,
-                blas::Op::ConjTrans, blas::Diag::NonUnit, T(1.0), t, std::move(w));
+                blas::Op::ConjTrans, blas::Diag::NonUnit, T(1.0), t, w);
+}
+
+template <class Executor, Device device, class T>
+void gemmUpdateW2Start(Executor&& ex, hpx::future<matrix::Tile<T, device>> w,
+                       hpx::shared_future<matrix::Tile<const T, device>> c,
+                       hpx::future<matrix::Tile<T, device>> w2) {
+  hpx::dataflow(ex, matrix::unwrapExtendTiles(tile::gemm_o), blas::Op::ConjTrans, blas::Op::NoTrans,
+                T(1.0), w, c, T(0.0), std::move(w2));
```

Comment:
I suspect this part has been merged incorrectly, as apparently the changes just revert the fixes of the reviews of #339.